src/Ledger/GovernanceActions.lagda

```diff
@@ -86,34 +89,45 @@ Figure~\ref{defs:governance} defines several data types used to represent govern
   to non-backwards compatible updates of a network. In Cardano, we formalize the definition slightly more by calling any upgrade that
   would lead to \emph{more blocks} being validated a ``hard fork'' and force nodes to comply with the new protocol version, effectively
   obsoleting nodes that are unable to handle the upgrade.}
-\begin{code}[hide]
+\begin{figure*}[h]
+\begin{code}
+⟨_,_,_⟩ : Maybe ℚ → Maybe ℚ → Maybe ℚ → GovRole → Maybe ℚ
+⟨ q₁ , q₂ , q₃ ⟩ = λ { CC → q₁ ; DRep → q₂ ; SPO → q₃ }
```

Comment:
This should be private, in this case just put it under `threshold`'s `where`.

Also the notation as a tuple does not seem very intuitive in this case, which might decrease the readibility of the "voting matrix" below (which I believe to be the central piece of interest in this page).

How about a notation that actually mentions which role is voting?
```agda
CC∶ q₁ DRep∶ q₂ SPO∶ q₃ = λ { CC → q₁ ; DRep → q₂ ; SPO → q₃ }
```
The signature is of no interest and could be hidden (also make it polymorphic, no need for being specific here):
```agda
CC∶_DRep∶_SPO∶_ : ∀ {a} {A : Set a} → A → A → A → GovRole → A
```

Reply:
I've tried your `CC∶_DRep∶_SPO∶_` suggestion, but the extra characters make it quite noisy and hard to read. I'm thinking that maybe we could just go for a notation that emulates a proper table: `|_|_|_|`. We could then even do some LaTeX trickery to give it a header line in the figure so it's clear what's what. Obviously we should also explain that in the prose.

Reply:
This doesn't seem noisy to me to be honest:
```agda
   NoConfidence           → CC∶ noVote            DRep∶ vote P1      SPO∶ vote Q1
     (NewCommittee _ _ _)   → case ccThreshold' of λ where
     (just _)             → CC∶ noVote            DRep∶ vote P2a     SPO∶ vote Q2a
     nothing              → CC∶ noVote            DRep∶ vote P2b     SPO∶ vote Q2b
   (NewConstitution _ _)  → CC∶ vote ccThreshold  DRep∶ vote P3      SPO∶ noVote
   (TriggerHF _)          → CC∶ vote ccThreshold  DRep∶ vote P4      SPO∶ vote Q4
   (ChangePParams x)      → CC∶ vote ccThreshold  DRep∶ vote (P5 x)  SPO∶ noVote
   (TreasuryWdrl _)       → CC∶ vote ccThreshold  DRep∶ vote P6      SPO∶ noVote
   Info                   → CC∶ vote 2ℚ           DRep∶ vote 2ℚ      SPO∶ vote 2ℚ
```

Reply:
Maybe you have a higher pain threshold than me, I find this presentation quite overwhelming. I already dislike all the `vote/noVote` functions quite a bit.

I've made `⟨_,_,_⟩` private.